Microsoft.Toolkit.Win32/Microsoft.Toolkit.Wpf.UI.XamlHost/WindowsXamlHostBase.cs

```diff
@@ -0,0 +1,184 @@
+                    frameworkElement.DataContext = DataContext;
+                }
+
+                // Fire updated event
+                ChildChanged?.Invoke(this, new EventArgs());
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this wrapper control instance been disposed
+        /// </summary>
+        protected bool IsDisposed { get; set; }
+
+        /// <summary>
+        /// Creates <see cref="Windows.UI.Xaml.Application" /> object, wrapped <see cref="Windows.UI.Xaml.Hosting.DesktopWindowXamlSource" /> instance; creates and
+        /// sets root UWP XAML element on <see cref="Windows.UI.Xaml.Hosting.DesktopWindowXamlSource" />.
+        /// </summary>
+        /// <param name="hwndParent">Parent window handle</param>
+        /// <returns>Handle to XAML window</returns>
+        protected override HandleRef BuildWindowCore(HandleRef hwndParent)
+        {
+            // 'EnableMouseInPointer' is called by the WindowsXamlManager during initialization. No need
+            // to call it directly here.
+
+            // Create DesktopWindowXamlSource instance
+            var desktopWindowXamlSourceNative = _xamlSource.GetInterop();
+
+            // Associate the window where UWP XAML will display content
+            desktopWindowXamlSourceNative.AttachToWindow(hwndParent.Handle);
+
+            var windowHandle = desktopWindowXamlSourceNative.WindowHandle;
+
+            // Overridden function must return window handle of new target window (DesktopWindowXamlSource's Window)
+            return new HandleRef(this, windowHandle);
+        }
+
+        /// <summary>
+        /// The default implementation of SetContent applies ChildInternal to desktopWindowXamSource.Content.
+        /// Override this method if that shouldn't be the case.
+        /// For example, override if your control should be a child of another WindowsXamlHostBase-based control.
+        /// </summary>
+        protected virtual void SetContent(Windows.UI.Xaml.UIElement newValue)
+        {
+            if (_xamlSource != null)
+            {
+                _xamlSource.Content = newValue;
+            }
+        }
+
+        /// <summary>
+        /// WPF framework request to destroy control window.  Cleans up the HwndIslandSite created by DesktopWindowXamlSource
+        /// </summary>
+        /// <param name="hwnd">Handle of window to be destroyed</param>
+        protected override void DestroyWindowCore(HandleRef hwnd)
+        {
+            Dispose(true);
+        }
+
+        /// <summary>
+        /// WindowsXamlHost Dispose
+        /// </summary>
+        /// <param name="disposing">Is disposing?</param>
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing && !IsDisposed)
+            {
+                IsDisposed = true;
+
+                _xamlSource.TakeFocusRequested -= OnTakeFocusRequested;
```

Comment:
`DesktopWindowXamlSource` is of type `IDisposable` and not disposed here as expected